cns/nmagentclient/nmagentclient.go

```diff
@@ -86,3 +96,42 @@ func GetNetworkContainerVersion(
 		networkContainerID, response, err)
 	return response, err
 }
+
+// GetNmAgentSupportedApis :- Retrieves Supported Apis from NMAgent
+func GetNmAgentSupportedApis(
+	getNmAgentSupportedApisURL string) ([]string, string) {
+	var (
+		returnMessage string
+		supportedApis []string
+	)
+
+	if getNmAgentSupportedApisURL == "" {
+		getNmAgentSupportedApisURL = fmt.Sprintf(
+			GetNmAgentSupportedApiURLFmt, WireserverIP)
+	}
+
+	response, err := common.GetHttpClient().Get(getNmAgentSupportedApisURL)
+	if err != nil || response.StatusCode != http.StatusOK || response == nil {
+		returnMessage = fmt.Sprintf(
+			"Failed to retrieve Supported Apis from NMAgent with error %v",
+			err.Error())
+		logger.Errorf("[Azure-CNS] %s", returnMessage)
```

Comment:
Seems like you can just let the caller code log the errors instead of logging before every return